### DISCUSSION TITLE
onClick  callback is not effective in renderSubLayers

### DISCUSSION THREAD

**Discussion by sanbai0** [visgl/deck.gl#6959] [Q&A] [answered]
Onclick callback works effectively outside renderSubLayers as follows：
`      const mvtLayer = new MVTLayer({
        data: data,
        pickable: true,
        onClick:(info, event) => console.log('Clicked:', info, event),
        renderSubLayers: (props) => {
          return new BaseCompositeLayer({
            id: props.id,
            data: data,
            pickable:true,
          });
        },
      });`

But onClick callback is not effective in renderSubLayers（No output after clicking） as follows：
`      const mvtLayer = new MVTLayer({
        data: data,
        pickable: true,
        renderSubLayers: (props) => {
          return new BaseCompositeLayer({
            id: props.id,
            data: data,
            pickable:true,
            onClick:(info, event) => console.log('Clicked:', info, event),
          });
        },
      });`

Is this normal？

**Comment by Pessimistress** [comment, marked as answer]
`onClick` is only fired on the top-level layer.
